galley/cmd/galley/cmd/root.go

```diff
@@ -48,21 +63,96 @@ func GetRootCmd(args []string, printf, fatalf shared.FormatFn) *cobra.Command {
 			if len(args) > 0 {
 				return fmt.Errorf("%q is an invalid argument", args[0])
 			}
-
 			return log.Configure(loggingOptions)
 		},
+		Run: func(cmd *cobra.Command, args []string) {
+
+			sa.KubeConfig = flags.kubeConfig
+			sa.ResyncPeriod = flags.resyncPeriod
+			sa.CACertificateFile = vc.CACertFile
+			sa.KeyFile = vc.KeyFile
+			sa.CertificateFile = vc.CertFile
+			noServer := flags.noServer
+			noValidation := flags.noValidation
+
+			if noServer && noValidation {
+				fatalf("Galley must be running under at least one mode: server or validation")
+			}
+
+
+			if !noServer {
+				go server.RunServer(sa, printf, fatalf)
+			}
+			if !noValidation {
+				go validation.RunValidation(vc, printf, fatalf, flags.kubeConfig)
+			}
+
+			//Run liveness and readiness check
+			probeCheckController := galleyProbe.NewProbeCheckController(flags.livenessProbeOptions,flags.readinessProbeOptions)
+			probeCheckController.Run()
+			galleyStop := make(chan struct{})
```

Comment:
Should we propagate the `galleyStop` channel to `server.RunServer` and `validation.RunValidation`?

Reply:
in server.RunServer and validation.RunValidation they will call os.Exit(-1) when they encounter errors. So I think we can use this channel to exit if none of them has errors.